sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -878,7 +919,7 @@ func getUpdatedContainerEndpoints(hostname string, networkInfo *sensor.NetworkCo
 				IPAndPort: getIPAndPort(endpoint.GetListenAddress()),
 				L4Proto:   net.L4ProtoFromProtobuf(endpoint.GetProtocol()),
 			},
-			processKey: getProcessKey(endpoint.GetOriginator()),
+			processKey: getProcessKey(endpoint.GetOriginator(), processes),
```

Comment:
This was a bug previously as ep was being used as a key, but due to the use of a pointer the same endpoint would appear multiple times in the map.

Reply:
Wouldn't passing by value instead passing by pointer reference solve this problem?

Reply:
I changed `processKey` to be of type `processInfo`.

Reply:
Using the `processes` map could reduce memory if one process opened many endpoints. Then you would just have one string as the key and one pointer to `processInfo` in the map, and each containerEndpoint would have the same pointer instead of having a `procesInfo` for each containerEndpoint. It could save memory or use more of it depending upon how many endpoints the average process has. Anyway it complicates things and probably uses more memory in most cases.